migrations/versions/660_move_framework_framework_agreement_version.py

```diff
@@ -0,0 +1,37 @@
+"""Move Framework.framework_agreement_version into new framework_agreement_details json field
+
+Revision ID: 660
+Revises: 650
+Create Date: 2016-06-29 11:54:27.719038
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '660'
+down_revision = '650'
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+def upgrade():
+    op.add_column('frameworks', sa.Column('framework_agreement_details', postgresql.JSON(), nullable=True))
+    op.execute("""
+        UPDATE frameworks SET
+            -- using to_json to ensure framework_agreement_version gets properly escaped for json
+            framework_agreement_details = (
+                '{"frameworkAgreementVersion":' || to_json(framework_agreement_version) || '}'
+            )::json
+        WHERE
+            framework_agreement_version IS NOT NULL and framework_agreement_version != ''
+    """)
+    op.drop_column('frameworks', 'framework_agreement_version')
+
+
+def downgrade():
+    op.add_column('frameworks', sa.Column('framework_agreement_version', sa.VARCHAR(), autoincrement=False, nullable=True))
+    op.execute("""
+        UPDATE frameworks SET
+            framework_agreement_version = framework_agreement_details->'frameworkAgreementVersion'
```

Comment:
Yes I see what it was - I was just seeing the quotes in the listing and thinking "yup thats how postgres shows a string..." - should have been kicked out of the postgres conference yesterday with that kind of thinking. Doing the `->>`.